### PR TITLE
draw option: vertical_compression: 'high', 'medium', 'low'

### LINKED ISSUE #2486: Text drawer option vertically_compressed is not exposed into higher drawing APIs
_Originally posted by @1ucian0 in https://github.com/Qiskit/qiskit-terra/pull/2480#issuecomment-494994853_

The text drawer has an option to enable and disable vertical compression: `vertically_compressed`. However, this option is not exposed by `circuit_drawer` or `QuantumCircuit.draw`.

### PR DESCRIPTION
**Pull request by 1ucian0**
Fixed #2486 

I could not sleep on the plane, so I added `vertical_compression` to address #2486. It looks like this:

```
qr = QuantumRegister(3, 'qr')
cr = ClassicalRegister(1, 'cr')
circuit = QuantumCircuit(qr, cr)
circuit.ccx(qr[0], qr[1], qr[2]).c_if(cr, 1)

print(circuit.draw(vertical_compression='high'))
print(circuit.draw(vertical_compression='low'))
print(circuit.draw(vertical_compression='medium'))
```
```
                
qr_0: |0>───■───
            │   
qr_1: |0>───■───
          ┌─┴─┐ 
qr_2: |0>─┤ X ├─
         ┌┴─┴─┴┐
 cr_0: 0 ╡ = 1 ╞
         └─────┘
                
qr_0: |0>───■───
            │   
            │   
qr_1: |0>───■───
            │   
          ┌─┴─┐ 
qr_2: |0>─┤ X ├─
          └─┬─┘ 
         ┌──┴──┐
 cr_0: 0 ╡ = 1 ╞
         └─────┘
                
qr_0: |0>───■───
            │   
qr_1: |0>───■───
          ┌─┴─┐ 
qr_2: |0>─┤ X ├─
          └─┬─┘ 
         ┌──┴──┐
 cr_0: 0 ╡ = 1 ╞
         └─────┘
```

The default is `medium`.

In detail, this PR:
 - renames `vertically_compressed` -> `vertical_compression`
 - exposes the option to `draw` and `circuit_drawer`.
 - extends the docstrings
 - choose the default in the lower API level to `high` to keep the tests

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -19,6 +19,12 @@ The format is based on `Keep a Changelog`_.
 `UNRELEASED`_
 =============
 
+Added
+-----
+
+- The option `vertical_compression` to the text drawer allows to control how much
+  room the text circuit drawing takes.
+
```
Comment (mtreinish) [inline CHANGELOG.rst]: We should mention that we added the parameter to the top level argument too

File: qiskit/circuit/quantumcircuit.py [hunk 2]
```diff
@@ -555,7 +555,9 @@ def draw(self, scale=0.7, filename=None, style=None, output=None,
                 gates should be placed in the output circuit if there is an option.
                 `none` results in each gate being placed in its own column. Currently
                 only supported by text drawer.
-
+            vertical_compression (string): `high`, `medium` or `low`. It merges the
+                    lines generated by `text` so the drawing will take less vertical
+                    room. Default is `medium`.
```
Comment (mtreinish) [inline qiskit/circuit/quantumcircuit.py]: We should mention that this is only for the text drawer here

File: CHANGELOG.rst (hunk 1 above)
Comment (1ucian0) [reply CHANGELOG.rst]: Fixed in c290fc64

File: qiskit/visualization/circuit_visualization.py [hunk 3]
```diff
@@ -93,7 +94,9 @@ def circuit_drawer(circuit,
             gates should be placed in the output circuit if there is an option.
             `none` results in each gate being placed in its own column. Currently
             only supported by text drawer.
-
+        vertical_compression (string): `high`, `medium` or `low`. It merges the
+                    lines generated by `text` so the drawing will take less
+                    vertical room. Default is `medium`.
```
[the remark this answers is not in the document]
Comment (1ucian0) [reply qiskit/visualization/circuit_visualization.py]: Fixed in a2aed414

File: qiskit/circuit/quantumcircuit.py (hunk 2 above)
Comment (1ucian0) [reply qiskit/circuit/quantumcircuit.py]: fixed in 22d37d2f

File: CHANGELOG.rst [hunk 4]
```diff
@@ -28,6 +34,8 @@ Changed
 - When adding a register to a circuit, an error will now be raised if a register
   of the same name is already present. Previously, an error would only be raised
   if the same register was added twice.
+- Qubits and classical bits are not represented as a tuples anymore, but as
+  instances of ``Qubit`` and ``Clbit`` respectively.
```
[the remark this answers is not in the document]
Comment (1ucian0) [reply CHANGELOG.rst]: fixed in abd1a25d

File: qiskit/visualization/circuit_visualization.py [hunk 5]
```diff
@@ -178,6 +181,8 @@ def circuit_drawer(circuit,
         creglinestyle (str): The style of line to use for classical registers.
             Choices are `'solid'`, `'doublet'`, or any valid matplotlib
             `linestyle` kwarg value. Defaults to `doublet`(`mpl` only)
+        vertical_compression (str): `high`, `medium`, or `low`. It merges the lines
+                    so the drawing will take less vertical room. Default is `high`.
```
[the remark this answers is not in the document]
Comment (1ucian0) [reply qiskit/visualization/circuit_visualization.py]: The style dict *is only used by the `mpl`, `latex`, and `latex_source`*. It's just a parameter.

[the remark this answers is not in the document]
Comment (mtreinish) [reply qiskit/visualization/circuit_visualization.py]: Right that's what I thought which is why I pointed this out, this section of the docstring is for fields in the style dict, so we should remove this addition. Sorry I wasn't super clear about that before.

[the remark this answers is not in the document]
Comment (1ucian0) [reply qiskit/visualization/circuit_visualization.py]: oh! I got you. Good catch. Fixed in 82a7afd1

File: CHANGELOG.rst [hunk 6]
```diff
@@ -29,13 +29,14 @@ Deprecated
...
+- The option ``vertical_compression`` was added to the text drawer and to the
+  ``QuantumCircuit.draw`` method. The option allows to control how much
+  room the text circuit drawing takes.
 - Introduced a visualization for the Pass Manager. (#2445)
 
 Changed
 -------
 
-- Qubits and classical bits are not represented as a tuples anymore, but as
-  instances of ``Qubit`` and ``Clbit`` respectively.
```
Comment (mtreinish) [inline CHANGELOG.rst]: Looks like another rebase mistake, I don't think we want to remove this.

File: qiskit/visualization/text.py [hunk 7]
```diff
@@ -442,14 +442,14 @@ class TextDrawing():
...
-                 line_length=None, vertically_compressed=True):
+                 line_length=None, vertical_compression='high'):
         self.qregs = qregs
         self.cregs = cregs
         self.instructions = instructions
 
         self.plotbarriers = plotbarriers
         self.line_length = line_length
-        self.vertically_compressed = vertically_compressed
+        self.vertical_compression = vertical_compression
```
Comment (mtreinish) [inline qiskit/visualization/text.py]: I think we should we have a check here to validate that the input is an allowed value? Just something like:

```suggestion
        if vertical_compression not in ['high', 'medium', 'low']:
            raise ValueError("Vertical compression can only be 'high', 'medium', or 'low'")
        self.vertical_compression = vertical_compression
```

File: CHANGELOG.rst (hunk 6 above)
Comment (1ucian0) [reply CHANGELOG.rst]: the entry seems twice in master

Comment (mtreinish) [reply CHANGELOG.rst]: Oh, I didn't realize it was duplicated there. This is fine then. Normally I'd say lets do this in a separate commit to keep with 1 logical change per commit, but the github workflow makes iterating like this more difficult.

**Pull request merged**